pkg/wasm/httpfetcher.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wasm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+const (
+	defaultRetryBackoff = 50 * time.Millisecond
+)
+
+// HTTPFetcher fetches remote wasm module with HTTP get.
+type HTTPFetcher struct {
+	defaultClient *http.Client
+	retryBackoff  time.Duration
+}
+
+// NewHTTPFetcher create a new HTTP remote wasm module fetcher.
+func NewHTTPFetcher() *HTTPFetcher {
+	return &HTTPFetcher{
+		defaultClient: &http.Client{
```

Comment:
I don't quite remember, does this follow re-directs?

Reply:
Yes it follows redirect: https://golang.org/pkg/net/http/#Client.Get